src/WebJobs.Script/Description/DotNet/DotNetFunctionDescriptorProvider.cs

```diff
@@ -180,5 +190,36 @@ protected override IFunctionInvoker CreateFunctionInvoker(string scriptFilePath,
             methodAttributes.Clear();
             return base.GetFunctionParameters(functionInvoker, functionMetadata, triggerMetadata, methodAttributes, inputBindings, outputBindings);
         }
+
+        internal static bool TryCreateReturnValueParameterDescriptor(Type functionReturnType, IEnumerable<FunctionBinding> bindings, out ParameterDescriptor descriptor)
+        {
+            descriptor = null;
+
+            var returnBinding = bindings.SingleOrDefault(p => p.Metadata.IsReturn);
+            if (returnBinding == null)
+            {
+                return false;
+            }
+
+            if (functionReturnType.IsGenericType && functionReturnType.GetGenericTypeDefinition() == typeof(Task<>))
```

Comment:
Should the check here be `functionReturnType is Task` instead? Then check if is generic? We may also want to consider failing with a descriptive message if we have a non-generic task.